src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -108,139 +164,124 @@ public override void ConfigureServices(IServiceCollection services)
 
         public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
-            var openIddictServerOptions = serviceProvider
-                .GetRequiredService<IOptionsMonitor<OpenIddictServerOptions>>()
-                .Get(OpenIddictServerDefaults.AuthenticationScheme);
+            async Task<OpenIdServerSettings> GetServerSettingsAsync()
+            {
+                // Note: the OpenID server service is registered as a singleton service and thus can be
+                // safely used with the non-scoped/root service provider available at this stage.
+                var service = serviceProvider.GetRequiredService<IOpenIdServerService>();
+
+                var configuration = await service.GetSettingsAsync();
+                if ((await service.ValidateSettingsAsync(configuration)).Any(result => result != ValidationResult.Success))
+                {
+                    return null;
+                }
+
+                return configuration;
+            }
 
-            if (openIddictServerOptions.AuthorizationEndpointPath != PathString.Empty)
+            var settings = GetServerSettingsAsync().GetAwaiter().GetResult();
+            if (settings == null)
             {
-                routes.MapAreaRoute(
```

Comment:
Looks like I missed that when reviewing the PR that introduced these routes, but the `Access.AuthorizeGet`/`Access.Deny` routes were buggy: neither `Accept` nor `Deny` are independent actions. Both actually point to the same `Authorize` action and we use `[FormValueRequired]` to select the right one at runtime.